pkg/get/get_test.go

```diff
@@ -470,7 +470,58 @@ func Test_DownloadInletsctl(t *testing.T) {
 			t.Fatal(err)
 		}
 		if got != tc.url {
-			t.Fatalf("want: %s, got: %s", tc.url, got)
+			t.Logf("want: %s, got: %s", tc.url, got)
+			t.Fail()
+		}
+	}
+}
+
+func Test_DownloadDoctl(t *testing.T) {
+	tools := MakeTools()
+	name := "doctl"
+
+	var tool *Tool
+	for _, target := range tools {
+		if name == target.Name {
+			tool = &target
+			break
+		}
+	}
+
+	type test struct {
+		os      string
+		arch    string
+		version string
+		url     string
+	}
+
+	const toolVersion = "1.46.0"
+	const urlPrefix = "https://github.com/digitalocean/doctl/releases/download"
+
+	tests := []test{
+		{os: "mingw64_nt-10.0-18362",
+			arch:    arch64bit,
+			version: toolVersion,
+			url:     urlPrefix + "/v" + toolVersion + "/doctl-1.46.0-windows-amd64.zip"},
+		{os: "linux",
+			arch:    arch64bit,
+			version: toolVersion,
+			url:     urlPrefix + "/v" + toolVersion + "/doctl-1.46.0-linux-amd64.tar.gz"},
+		{os: "darwin",
+			arch:    arch64bit,
+			version: toolVersion,
+			url:     urlPrefix + "/v" + toolVersion + "/doctl-1.46.0-darwin-amd64.tar.gz"},
+	}
+
+	for _, tc := range tests {
+		got, err := tool.GetURL(tc.os, tc.arch, tc.version)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if got != tc.url {
+			t.Logf("want: %s, got: %s", tc.url, got)
+			// this ensures that if one or more tests fail, it gets marked as failed, but we run others to see if they fail too
```

Comment:
@Waterdrips also, may I ask you to let me know what you think of this? My intent is to get all test failures reported at one time, rather than stop on the first failure and report back. Is there a preference?

Reply:
@Waterdrips is it okay that I'm using logf and fail here?